src/sknnr/transformers/_cca.py

```diff
@@ -193,7 +193,7 @@ def coefficients(self):
 
     def projector(self, n_components=None):
         if n_components is None:
-            return self.coefficients @ self.axis_weights
+            n_components = self.max_components
```

Comment:
I didn't notice until I ran test coverage and saw this is uncovered, but this condition won't ever happen during normal usage because `n_components` will always be passed *after* it goes through `ComponentReducerMixin.set_n_components`, which effectively runs the same check for `None` values. 

I'm fine leaving this in if you think there are possible cases where a user might want to access this directly, but if not then maybe we don't need the check at all and `n_components` can be a required param?

Reply:
Great question.  When I first wrote these classes, it was basically code dumps for the `CCA` and `CCorA` classes from stuff I had already done and then `CCATransformer` and `CCorATransformer` were meant to be thin layers on top to conform to scikit-learn.  While one could certainly call `CCA` or `CCorA` directly, I don't think that would be a typical use case.  For now, I like your suggestion and will make this a required param.  If it comes back to bite us, it's an easy transition back to this form.

Reply:
Cool, makes sense to me!